dotnet/src/Connectors/Connectors.OpenAI/OpenAIPromptExecutionSettings.cs

```diff
@@ -254,6 +254,39 @@ public int ResultsPerPrompt
         }
     }
 
+    /// <summary>
+    /// Whether to return log probabilities of the output tokens or not.
+    /// If true, returns the log probabilities of each output token returned in the `content` of `message`.
+    /// </summary>
+    [Experimental("SKEXP0010")]
+    [JsonPropertyName("logprobs")]
+    public bool? EnableLogProbabilities
+    {
+        get => this._enableLogProbabilities;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._enableLogProbabilities = value;
+        }
+    }
+
+    /// <summary>
+    /// An integer specifying the number of most likely tokens to return at each token position, each with an associated log probability.
+    /// </summary>
+    [Experimental("SKEXP0010")]
+    [JsonPropertyName("top_logprobs")]
+    public int? LogProbabilitiesCount
+    {
+        get => this._logProbabilitiesCount;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._logProbabilitiesCount = value;
+        }
+    }
```

Comment:
What about the consumption of the logprobs that come back in the response? Does that already work and they're just surfaced somewhere in the Metadata?

Reply:
Yes, today they are already included in response metadata:
https://github.com/microsoft/semantic-kernel/blob/33f278c477c264aa1e84caedd9e41cd1ea926582/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L248

https://github.com/microsoft/semantic-kernel/blob/33f278c477c264aa1e84caedd9e41cd1ea926582/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L268

Reply:
Added integration test that verifies consumption behavior.

Reply:
Cool, thanks.